### ISSUE TITLE
AssertionError in train MOT20

### ISSUE THREAD

**Issue by iTruffle** [noahcao/OC_SORT#57]
I can successfully run evaluation code for MOT17 and MOT20 and video demo.
But I got an error in training for MOT20, using 'python tools/train.py -f exps/example/mot/yolox_x_mix_mot20_ch.py -d 1 -b 48 --fp16 -o -c pretrained/yolox_x.pth', like this:
![image](https://user-images.githubusercontent.com/80766557/180768539-18dbc670-ea89-427a-ab58-5800dd7a4eee.png)

**Comment by iTruffle** [comment]
And I have followed the instruction to change the code in 'line 134-135 in data_augment.py, line 122-125 in mosaicdetection.py, line 217-225 in mosaicdetection.py, line 115-118 in boxes.py' .
After challenging, the code like this:
![image](https://user-images.githubusercontent.com/80766557/180770976-ad703382-15bc-4791-8838-4abca37a1649.png)
![image](https://user-images.githubusercontent.com/80766557/180770993-0e158741-dd61-418c-a6a0-69d934e80677.png)
![image](https://user-images.githubusercontent.com/80766557/180771103-032fc9ea-0428-4b66-b8c2-3e791b960035.png)
![image](https://user-images.githubusercontent.com/80766557/180771132-fd02a3b0-6ed7-4922-a1bf-a85f6d57a3cd.png)

**Comment by noahcao** [comment]
This is possible if you read an empty file or in a wrong path name. I tried the code on my side, it looks right. Have you fixed this issue?

**Comment by iTruffle** [comment]
Sorry, I do not know where is wrong and how to train successfully.

**Comment by YamacTan** [comment]
I faced with the same problem today. After examining the error logs, I saw that although the annotation files were created, there were no related images in the "mix_mot" tag or other similar folders created. So because the folders are empty it can't find the images and this error occurs.
After adding the images to the folders manually, the error was resolved and training started. I'm not sure if this is a bug as I haven't had a chance to review the scripts that mix the data yet.

I hope it helps.
Y.
